game/PsyDoom/Launcher/Launcher_Tab_Multiplayer.cpp

```diff
@@ -0,0 +1,128 @@
+    }
+
+    // Enable multiplayer-only things
+    {
+        const auto pCheck = makeFl_Check_Button(x + 20, y + 70, 150, 30, "  Spawn Multiplayer-Only Things");
+        bindConfigField<Config::gbMPThings, Config::gbNeedSave_Multiplayer>(*pCheck);
+        pCheck->tooltip(ConfigSerialization::gConfig_Multiplayer.mpThings.comment);
+    }
+
+        // Preserve keys on respawn
+    {
+        const auto pCheck = makeFl_Check_Button(x + 20, y + 100, 150, 30, "  Preserve Keys on Respawn");
+        bindConfigField<Config::gbPreserveKeys, Config::gbNeedSave_Multiplayer>(*pCheck);
+        pCheck->tooltip(ConfigSerialization::gConfig_Multiplayer.preserveKeys.comment);
+    }
+
+        // Preserve ammo on respawn
+    {
+        const auto pLabel = new Fl_Box(FL_NO_BOX, x + 20, y + 140, 120, 26, "Preserve Ammo on Respawn");
+        pLabel->align(FL_ALIGN_LEFT | FL_ALIGN_INSIDE);
+        pLabel->tooltip(ConfigSerialization::gConfig_Multiplayer.preserveAmmoFactor.comment);
+
+        const auto pChoice = new Fl_Choice(x + 210, y + 140, 100, 26);
+        pChoice->add("None");
+        pChoice->add("All");
+        pChoice->add("Half");
+        pChoice->tooltip(pLabel->tooltip());
+        pChoice->callback(
+            [](Fl_Widget* const pWidget, void*) noexcept {
+                Fl_Choice* const pChoice = static_cast<Fl_Choice*>(pWidget);
+
+                switch (pChoice->value()) {
+                    case 0: Config::gPreserveAmmoFactor = 0; break;
+                    case 1: Config::gPreserveAmmoFactor = 1; break;
+                    case 2: Config::gPreserveAmmoFactor = 2; break;
+                }
+
+                Config::gbNeedSave_Multiplayer = true;
+            }
+        );
+
+        if (Config::gPreserveAmmoFactor == 2) {
+            pChoice->value(2);
+        } else if (Config::gPreserveAmmoFactor == 1) {
+            pChoice->value(1);
+        } else {
+            pChoice->value(0);
+        }
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+// Makes the 'deathmatch' section
+//------------------------------------------------------------------------------------------------------------------------------------------
+static void makeDeathmatchSection(const int x, const int y) noexcept {
+    // Container frame
+    new Fl_Box(FL_NO_BOX, x, y, 336, 30, "Deathmatch");
+    new Fl_Box(FL_THIN_DOWN_BOX, x, y + 30, 336, 100, "");
+
+        // Frag limit
+    {
+        const auto pLabel = new Fl_Box(FL_NO_BOX, x + 20, y + 50, 140, 26, "Frag Limit");
+        pLabel->align(FL_ALIGN_LEFT | FL_ALIGN_INSIDE);
+        pLabel->tooltip(ConfigSerialization::gConfig_Multiplayer.fragLimit.comment);
+
+        const auto pInput = new Fl_Int_Input(x + 200, y + 50, 110, 26);
+        bindConfigField<Config::gFragLimit, Config::gbNeedSave_Multiplayer>(*pInput);
```

Comment:
I saw some screenshots on doomworld where the space looked a little tight between some of the UI elements:
![image](https://user-images.githubusercontent.com/3496135/198515827-e0f87dd6-84de-47c8-aa1a-7f1c6fdb9e33.png)

Please make sure to add some extra spacing (maybe enough for 5 characters or so) to account for font rendering differences on Linux and macOS. The fonts on both those OSes can take up more room horizontally, which can cause overlaps if the layout is already tight on Windows. Also, we should try and allow a little padding between elements if they are close. In this case we have plenty of room so should be no problem to expand out a little horizontally...

Reply:
Apologies for this interjection but I noticed that there is a "Spawn Multiplayer-Only Things" option here. That almost gives the game an game Ultra Violence+ difficulty mode but still missing the Nightmare difficulty monster speed. I don't wish to labor the point or hijack this discussion as I know it was previously rejected, but could such a checkbox option be added to the main Launcher tab? Enable Ultra Violence+ difficulty with a warning stating (Passwords not enabled saves only) Also while on the game title screen it could state the difficulty too "UV Plus". This would be for verification purposes if people are doing a longplay or a run. Right now you could just use Nightmare with "Spawn Multiplayer-Only Things" and it would be UV Plus but being able to at least show you are in this difficulty mode from the title screen would be welcomed. I just want to state the reason I'm in part passionate about this difficulty mode as I'm looking to convert some WADs when I have that time and they make heavy use of extra multiplayer monsters and it doesn't look like multiplayer will operate over the internet anytime soon so If I'm going to play solo I'd like to experience it all.

Reply:
> Apologies for this interjection but I noticed that there is a "Spawn Multiplayer-Only Things" option here. That almost gives the game an Ultra Violence+ difficulty mode but still missing the Nightmare difficulty monster speed. I don't wish to labor the point or hijack this discussion as I know it was previously rejected, but could such a checkbox option be added to the main Launcher tab? Enable Ultra Violence+ difficulty with a warning stating (Passwords not enabled saves only) Also while on the game title screen it could state the difficulty too "UV Plus". This would be for verification purposes if people are doing a longplay or a run. Right now you could just use Nightmare with "Spawn Multiplayer-Only Things" and it would be UV Plus but being able to at least show you are in this difficulty mode from the title screen would be welcomed. I just want to state the reason I'm in part passionate about this difficulty mode is I'm looking to convert some WADs when I have that time and they make heavy use of extra multiplayer monsters and it doesn't look like multiplayer will operate over the internet anytime soon so If I'm going to play solo I'd like to experience it all.

See my comment here:
https://github.com/BodbDearg/PsyDoom/pull/64#discussion_r1009852068

Reply:
814e7965b4b13424d9f1072bb6c9ed2a938b87e6